modules/invoice/view/ajax.php

```diff
@@ -50,4 +50,282 @@
+  );
+  if ($line_control_total_differs->TotalDiffers) {
+    $messages .= "Total differs";
+  } else {
+    // TOTAL Matches
+    array_push($voucher_ids, $line_control_total_differs->VoucherID);
+    unset($journal_vouchers[$line_control_total_differs->VoucherID]);
+  }
+
+  // Check if lines and their VAT differ
+  $lines_differ = false;
+  $vat_lines_differ = false;
+  $query_invoice_lines = "
+    SELECT
+      il.*,
+      ROUND(
+        (
+          SELECT
+            IFNULL(
+              SUM(
+                IF(
+                  ilac.ChargeIndicator = 1,
+                  ilac.Amount,
+                  -ilac.Amount
+                )
+              ),
+              0
+            )
+          FROM
+            invoicelineallowancecharge ilac
+          WHERE
+            ilac.AllowanceChargeType = 'line' AND
+            ilac.InvoiceType = 'out' AND
+            ilac.InvoiceLineID = il.LineID
+        ),
+        2
+      ) AS InvoiceLineAllowanceChargeTotal
+    FROM
+      invoiceoutline il
+    WHERE
+      il.Active = 1 AND
+      il.InvoiceID = $InvoiceID
+    ";
+  $result_invoice_lines = $_lib['db']->db_query($query_invoice_lines);
+  while($invoice_line = $_lib['db']->db_fetch_object($result_invoice_lines)){
+    $line_amount = round($invoice_line->QuantityDelivered * $invoice_line->UnitCustPrice + $invoice_line->InvoiceLineAllowanceChargeTotal, 2);
+    $line_vat_amount = round($invoice_line->Vat/100 * $line_amount, 2);
+    $line_amount = round($line_amount + $line_vat_amount, 2);
+    $query_line = "
+      SELECT
+        *
+      FROM
+        voucher v
+        JOIN
+        accountplan ap
+        ON
+          ap.AccountPlanID = v.AccountPlanID
+      WHERE
+        v.VoucherID NOT IN (" . implode(', ', $voucher_ids) . ") AND
+        ap.EnableReskontro = 0 AND
+        (
+          v.AmountIn = $line_amount OR
+          v.AmountOut = $line_amount
+        ) AND
+        v.JournalID = $InvoiceID AND
+        v.VoucherType = 'S' AND
+        v.Active = 1
+    ";
+    $line = $_lib['db']->get_row(
+      array(
+        'query' => $query_line
+      )
+    );
+    if ($line) {
+      // LINE Matches
+      array_push($voucher_ids, $line->VoucherID);
+      unset($journal_vouchers[$line->VoucherID]);
+    } else {
+      $lines_differ = true;
+    }
+    $query_vat_lines = "
+      SELECT
+        *
+      FROM
+        voucher v
+        JOIN
+        accountplan ap
+        ON
+          ap.AccountPlanID = v.AccountPlanID
+      WHERE
+        v.VoucherID NOT IN (" . implode(', ', $voucher_ids) . ") AND
+        ap.EnableReskontro = 0 AND
+        (
+          v.AmountIn = $line_vat_amount OR
+          v.AmountOut = $line_vat_amount
+        ) AND
+        v.JournalID = $InvoiceID AND
+        v.VoucherType = 'S' AND
+        v.Active = 1
+    ";
+    $result_vat_lines = $_lib['db']->db_query($query_vat_lines);
+    while($vat_line = $_lib['db']->db_fetch_object($result_vat_lines)){
+      if ($vat_line) {
```

Comment:
this will include all the vat lines of the same amount even for the other lines that have the same amount, see how to fix this